lib/load-session-from-socket.js

```diff
@@ -30,14 +30,18 @@ module.exports = function loadSessionFromSocket(socketHandshake, app, cb) {
 
   // If no cookie exists, generate one and set it on the handshake
   var hadToGenerateCookie;
-  if (!socketHandshake.headers.cookie){
-    socketHandshake.headers.cookie = app.session.generateNewSidCookie();
+  // Check if there is no cookie, or a cookie is present and it doesn't have sails.sid
+  if (!socketHandshake.headers.cookie || ('string' === typeof socketHandshake.headers.cookie &&
+    !~socketHandshake.headers.cookie.indexOf(app.config.session && app.config.session.name || 'sails.sid'))){
+    var oneTimeCookie = app.session.generateNewSidCookie();
+
+    socketHandshake.headers.cookie = [socketHandshake.headers.cookie, oneTimeCookie].filter(Boolean).join('; ') ;
```

Comment:
I’m sure this wouldn’t be a primary code path, but I also hope this is performing enough.

Reply:
@shamasis what do you suggest here?

Reply:
Well nothing too drastic. I would do this too. But perhaps spent some time to see if there are performant alternative. For now let’s go with this.